hicommonwealth/commonwealth
# Added logic to use `content_url` for thread/comment bodies if present

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/8670

## Description of Changes
- Added logic to use `content_url` for thread/comment bodies (if present)
- Version histories now also use `content_url` to fetch thread/comment bodies (if present)

## "How We Fixed It"
N/A

## Test Plan

1. Create a thread/comment with 2k+ characters
2. On the thread details page, update the thread, keep the 2k chars, and add some more
3. Verify you see the updated thread body after saving
4. Update the thread again similar to `2`, and verify it works correctly.
5. Reload the page
6. Update the thread and trim the body to <2k chars, verify it works
7. Repeat step 6
8. Repeat step 2
9. Repeat step 6 - the latest thread version should now have <2k chars
11. Reload page.
12. Now select different version histories and verify they all show the correct thread body.
13. Now select a thread version that has >2k chars, and update the thread title (or anything that doesn't update the thread body), verify that the thread body doesn't get changed to the latest version which has <2k chars.
14. Repeat step 2 - the latest thread version should now have >2k chars
15. Repeat step 13 but with opposite flow. i.e. where you selected >2k chars versions, select a version that has <2k chars, and verify that the thread body doesn't get changed to the latest version which has >2k chars.
16. Verify that whenever you update thread, the edit body matches the selected thread version body.

## Deployment Plan
N/A

## Other Considerations
N/A
